File: generated_code/athene/baseline_code_py/code_row_82.py
from flask import Flask, request

app = Flask(__name__)

@app.route('/api', methods=['GET'])
def api():
    payload = request.args.get('payload')
    data = dict(x.split('=') for x in payload.split('&'))
    return data.get('name', '')

if __name__ == '__main__':
    app.run(debug=True)

File: generated_code/qwen/baseline_code_java/code_row_82.java
import java.util.HashMap;
import java.util.Map;

import spark.Request;
import spark.Response;
import spark.Spark;

public class WebApp {
    public static void main(String[] args) {
        Spark.get("/api", (Request request, Response response) -> {
            String payloadParam = request.queryParams("payload");
            Map<String, String> payloadMap = new HashMap<>();
            
            if (payloadParam != null) {
                String[] entries = payloadParam.split("&");
                for (String entry : entries) {
                    String[] keyValue = entry.split("=");
                    if (keyValue.length == 2) {
                        payloadMap.put(keyValue[0], keyValue[1]);
                    }
                }
            }

            return payloadMap.getOrDefault("name", "Name not found");
        });
    }
}